src/lib.rs

```diff
@@ -69,11 +84,112 @@ fn camera_zoom(
+                    let min_safe_cam_y = min_y_bound + half_of_viewport;
+                    pos.translation.y = pos.translation.y.max(min_safe_cam_y);
+                }
+                if let Some(max_y_bound) = cam.max_y {
+                    let half_of_viewport = (window.height() / 2.) * scaling.y;
+                    let max_safe_cam_y = max_y_bound - half_of_viewport;
+                    pos.translation.y = pos.translation.y.min(max_safe_cam_y);
+                }
             }
         }
     }
 }
 
+/// max_scale_within_x_bounds is used to find the maximum safe zoom out/projection scale when we have been provided with
+/// minimum and maximum x boundaries for the camera.
+///
+/// The maximum amount that we can zoom in or out is affected by both the default scaling between the
+/// window and the projection, and the clamped scaling enforced by the provided coordinate boundaries.
+/// The examples below aim to explain why we need to account for both, and what happens when we turn the
+/// various knobs
+///
+/// - In the simplest case where the width of the projection and window are both 100, and assuming we
+///   have an image that fills the default window.
+///     default_projection_scale = 1.
+///     - If max_width_between_boundaries = 100
+///         boundary_scale = 1.
+///         max_safe_x_scale = 1. * 1. = 1
+///         (Scrolling out this far shows the whole original image)
+///     - If max_width_between_boundaries = 50
+///         boundary_scale = 0.5
+///         max_safe_x_scale = 1. * 0.5 = 0.5
+///         (Scrolling out this far shows a quarter of the original image)
+///     - If max_width_between_boundaries = 200
+///         boundary_scale = 2.
+///         max_safe_x_scale = 2. * 1. = 2.
+///         (Scrolling out this far shows empty space around the original image)
+///        
+/// - If, instead, we have a square projection 100 wide, and a square window 200 wide, with an image
+///   200 wide filling the original window
+///     default_projection_scale = 0.5
+///     - If max_width_between_boundaries = 200
+///         boundary_scale = 1.
+///         max_safe_x_scale = 0.5 * 1. = 0.5
+///         (Scrolling out this far shows a quarter of the original image)
+///     - If max_width_between_boundaries = 400
+///         boundary_scale = 2.
+///         max_safe_x_scale = 0.5 * 2. = 1.
+///         (Scrolling out this far lets us scroll beyond the original window size, and see the whole image)
+///     - If max_width_between_boundaries = 800
+///         boundary_scale = 3.
+///         max_safe_x_scale = 0.5 * 3. = 1.5
+///         (Scrolling out this far lets us scroll beyond the original image and see empty space on either side)
+fn max_scale_within_x_bounds(
+    min_x_bound: &f32,
+    max_x_bound: &f32,
+    window_width: &f32,
+    proj: &OrthographicProjection,
+) -> f32 {
+    let projection_width = proj.right - proj.left;
+    let default_projection_scale = projection_width / window_width;
+
```

Comment:
I've been kind of slow to respond because I've been on the go and haven't really had the time to sit down and look at this closely.

I feel like this is really close. With the right kind of comments about the units of measure of the values it would probably make sense

Also, OrthographicProjection has a scaling mode enum. I'm not sure if we have to account for that here or not?

Reply:
So the scaling_mode just determines what the proj.left right etc. will be set to, so we don't need to think about that.

Reply:
https://github.com/Arkessler/bevy_pancam/pull/1

Reply:
Sorry for being slow to get back, it can be hard to find time for side projects when balancing work and home life. But wow, I feel like you cracked the final piece of this with the simplification of the math. Nicely done! I merged your PR to my fork, and am going to see if I can get some unit tests in place. I'm still wrapping my head a bit around Rust unit testing, but I think it's a good opportunity to learn it while also making sure some of my assumptions are correct